src/CleanSoftwareCron.php

```diff
@@ -82,6 +82,13 @@ public static function run(?int $max): int
     {
         $total = 0;
 
+        // Purge deleted links
+        $item_softwareVersion = new Item_SoftwareVersion();
+        $item_softwareVersion->deleteByCriteria([
+            'is_deleted' => 1,
+            'is_dynamic' => 1,
+        ]);
```

Comment:
With this, you will delete rows that are not necessary related to softwares that will be deleted on next line.

Updating criteria in `getVersionsWithNoInstallationCriteria` to exclude deleted `Item_SoftwareVersion` is a better solution.